internal/log/mock/logger.go

```diff
@@ -21,6 +21,43 @@ func (l *LoggerMock) Errorf(format string, args ...interface{}) {
 	return
 }
 
+// CopyWith mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) CopyWith(prefix string, isDebug bool) log.Logger {
+	return new(LoggerMock)
+}
+// DebugEnabled mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) DebugEnabled() bool {return false}
+// Prefix mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Prefix() string { return "" }
+// Debug mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Debug(...interface{}) {}
+// Debugf mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Debugf(string, ...interface{}) {}
+// Debugln mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Debugln(...interface{}) {}
+// Info mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Info(...interface{}) {}
+// Infof mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Infof(string, ...interface{}) {}
+// Infoln mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Infoln(...interface{}) {}
+// Warn mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Warn(...interface{}) {}
+// Warnf mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Warnf(string, ...interface{}) {}
+// Warnln mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Warnln(...interface{}) {}
+// Error mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Error(...interface{}) {}
+// Errorln mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Errorln(...interface{}) {}
+// Panic mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Panic(...interface{}) {}
+// Panicf mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Panicf(string, ...interface{}) {}
+// Panicln mocks the method of the same name on the log.Logger interface
+func (l *LoggerMock) Panicln(...interface{}) {}
+
```

Comment:
Can't you just embed the interface to avoid all this boilerplate.  Than any test using the mock would be over-riding specific methods itself.

Reply:
I don't know in advance which methods are being used in a test. If they're not implemented and they get use then that results in a panic. Might as-well have stub methods

Reply:
Hmmm... I'm not so sure.  I think it might be an advantage to force tests to be explicit.  Then again, I'm not sure it's wrong, either, so let's leave it but just keep an eye an that.